### PR TITLE
[feat]: inhibitory plasticity trace

### PR DESCRIPTION
**Pull request by and-rewsmith**
Adding an inhibitory plasticity trace. This was specified in the paper but we are still discussing if what is written was intended. Potential problems surrounding the inhibitory learning rule and trace from the paper:
1. Learning rule looks to be a symmetric hebbian profile, which I still have some lingering questions about. Namely how the decay works.
2. Typo in trace which we think should be entirely local to a given neuron. 

Regarding point 2,I have implemented this fixing the typo and keeping the trace local. The learning rule can now be implemented, but I will hold off until the dust settles around our discussion. We still have concerns over it being symmetric rather than antisymmetric.

### CODE REVIEW

File: model/src/util.py [hunk 1]
```diff
@@ -174,3 +176,31 @@ def tracked_value(self) -> torch.Tensor:
...
+        """
+        self.trace: Optional[torch.Tensor] = None
+        self.tau_stdp: float = tau_stdp
+
+    def apply(self, spike: torch.Tensor, dt: float = DT) -> torch.Tensor:
+        if self.trace is None:
+            # Initialize variance based on the first spike received
+            self.trace = torch.zeros_like(spike)
+
+        self.trace += (-1) * (dt / self.tau_stdp) * self.trace + spike
```
Comment (and-rewsmith) [inline model/src/util.py:202]: I want to call out that this is a different form than the others for moving averages that decay. Example where `dt/tau` is applied to the whole term:


https://github.com/and-rewsmith/LPL-SNN/blob/2a1466b4f5fb8bb4c76d2b7b03b9a417b3360289/model/src/util.py#L134

Comment (mstormbull) [reply model/src/util.py:202]: Agreed, I believe that this manifests itself as a tau/dt scale factor on the STDP rule. Do you see any other ramifications of this choice?

**Review by mstormbull — APPROVED** [review]
LGTM,

File: model/src/util.py (hunk 1 above)
Comment (and-rewsmith) [reply model/src/util.py:202]: > Agreed, I believe that this manifests itself as a tau/dt scale factor on the STDP rule. 

I have added this to the list of things to discuss next time we chat. Will need to understand this more.

**Pull request merged**